### PR TITLE
Swift 2.0

### PR DESCRIPTION
**Pull request by cojoj**
This is my attempt with support for Swift 2.0. 

I've changed some things but still there are more possibilities to change some logic. We don't have enough test cases to fully check if everything will work and I think this should be the first thing to do after slicing this SDK from Buildasaur.

@czechboy0 if you have some possibilities to test my changes without unit tests, please do so.

### CODE REVIEW

**Comment by czechboy0** [comment]
Great stuff, I'll take a look at it properly when I'm back in London (traveling atm). Thanks!

**Comment by cojoj** [comment]
A pleasure to contribute in Swift 😉 Have a safe flight!

One more thing to add - so far the conversion was done by hand. I had to do this as some of those things were to complicated for converter. As I suggested in #4 it'd be advisable to take the whole concept, rethink it and refactor bit by bit. Everything is on a good way!

**Comment by cojoj** [comment]
Some new stuff... 😜

I've created `.gitignore` file (wasn't there) and added some useful **ignores**. Also removed files from git's cache to exclude files from `.gitignore`.

I've also sketched the inheritance hierarchy and based on that I've grouped files to separate directories (mentioned in #3). Maybe it's not the perfect categorization but it's a good place to start.
Now it looks like:
- **XcodeServerSDK**
  - General files
  - Server Entities [dir]
  - Server Helpers [dir]
  - Utils [dir]

You'll see that in a root directory of the project there's directory called **Playgrounds**. I created it after some suggestions from WWDC movie so while testing some code you won't need to build the whole project 😉

From now on I won't add anything to this **PR**. I'll just sit there and wait...

**Comment by czechboy0** [comment]
Awesome. Feel free to push more stuff of you want though, I probably won't have time to review this until Tuesday.

**Comment by cojoj** [comment]
Due to integrating of playgrounds to this project I'd be nice to have #7 merged. Also it'd be advisable to create `.xcworkspace`. If those requirements were met we can easily use `import XcodeServerSDK` inside any playground ☺️

File: XcodeServerSDK/Utils/HTTPUtils.swift [hunk 1]
```diff
@@ -53,13 +53,13 @@ public class HTTP {
                         
                     case let s where s.rangeOfString("application/json") != nil:
                         
-                        let (json: AnyObject!, error) = JSON.parse(data)
-                        let headers = httpResponse.allHeaderFields
+                        let (json, error) = JSON.parse(data!)
+                        // let headers = httpResponse.allHeaderFields
```
Comment (sync) [inline XcodeServerSDK/Utils/HTTPUtils.swift:55]: avoid doing this, instead you should be doing this 

upper in the code replace

```
if data == nil {

                    //no body, but a valid response
                    completion(response: httpResponse, body: nil, error: nil)
                    return
                }
```

with this:

```
guard let data = data else {

                    //no body, but a valid response
                    completion(response: httpResponse, body: nil, error: nil)
                    return
                }
```

so after data data won't be an optional anymore so this `let (json, error) = JSON.parse(data!)` becomes `let (json, error) = JSON.parse(data)`

File: XcodeServerSDK/Utils/HTTPUtils.swift [hunk 2]
```diff
@@ -53,13 +53,13 @@ public class HTTP {
...
                         completion(response: httpResponse, body: json, error: error)
                         
                     default:
                         //parse as UTF8 string
-                        let string = NSString(data: data, encoding: NSUTF8StringEncoding) as! String
+                        let string = NSString(data: data!, encoding: NSUTF8StringEncoding) as! String
```
Comment (sync) [inline XcodeServerSDK/Utils/HTTPUtils.swift:61]: same here:

avoid doing this, instead you should be doing this 

upper in the code replace

```
if data == nil {

                    //no body, but a valid response
                    completion(response: httpResponse, body: nil, error: nil)
                    return
                }
```

with this:

```
guard let data = data else {

                    //no body, but a valid response
                    completion(response: httpResponse, body: nil, error: nil)
                    return
                }
```

so after data data won't be an optional anymore so this `let string = NSString(data: data!, encoding: NSUTF8StringEncoding) as! String` becomes `let string = NSString(data: data, encoding: NSUTF8StringEncoding) as! String`

File: XcodeServerSDK/Utils/HTTPUtils.swift (hunk 1 above)
Comment (czechboy0) [inline XcodeServerSDK/Utils/HTTPUtils.swift:55]: +1

File: XcodeServerSDK/Utils/HTTPUtils.swift (hunk 2 above)
Comment (czechboy0) [inline XcodeServerSDK/Utils/HTTPUtils.swift:61]: +1, thanks @sync, nice catches!

**Comment by czechboy0** [comment]
@cojoj Can you also please remake this PR into branch `swift-2` please? I'd like to always keep master on the latest stable Swift version (but obviously most development will go into the Swift 2 version going forward).

**Pull request closed without merging**